File: gateway/src/interfaces/auth/session_svc.go
package auth

import (
	"context"
	authmodel "gateway/src/models/auth"
)

// 服务模块侧（非认证中心）的会话管理服务接口，包含受限的会话访问方法。
type ISessionService interface {
	// GetSession 获取会话信息，sessionID 是会话标识符。
	GetSession(ctx context.Context, sessionID string) (*authmodel.Session, error)
	// TouchSession 更新会话信息，sessionID 是会话标识符，meta 是更新元数据。
	TouchSession(ctx context.Context, sessionID string, meta authmodel.SessionTouchMeta) error
	// ValidateSession 校验会话状态，用于出站调用前快速失败。
	ValidateSession(ctx context.Context, req *authmodel.SessionValidateRequest) (*authmodel.Session, error)
}

File: api_service/src/services/auth/session_svc.py
from uuid import UUID

from src.models.auth.auth import Session, SessionTouchMeta
from src.models.auth.auth_contract import SessionValidateRequest


class SessionService:
    """会话服务。运行期不回源认证中心。"""

    def __init__(self) -> None:
        pass

    def upsert_session(self, session: Session | None) -> None:
        # 非认证中心模块不再维护本地会话状态。
        _ = session

    async def upsert_session_async(self, session: Session | None) -> None:
        _ = session

    async def get_session(self, session_id: str) -> Session | None:
        try:
            UUID(session_id)
        except ValueError:
            return None
        return None

    async def touch_session(self, session_id: str, meta: SessionTouchMeta) -> None:
        # 触达元信息由认证中心统一管理；本模块不做本地触达更新。
        _ = (session_id, meta)

    async def validate_session(self, req: SessionValidateRequest) -> Session | None:
        _ = req
        return None
